Lib/tkinter/simpledialog.py

```diff
@@ -260,7 +258,7 @@ def __init__(self, title, prompt,
                  parent = None):
 
         if not parent:
-            parent = tkinter._default_root
+            parent = _get_default_root('create query dialog')
```

Comment:
I'm not sure this whole check for parent is needed as parent is ultimately passed to `Toplevel` which handles it.